From 5001a059e9c883746754ed6411a3d2587d0717fc Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Tue, 28 Aug 2018 17:04:29 +0530
Subject: [PATCH 0170/1051] soc: octeontx2: Avoid cgx fwi requests when lmac
 count is 0

When cgx doesn't have any lmac port initialized by firmware,
avoid firmware interface commands to that port.

Along with that remove a redundant EXPORT symbol.

Signed-off-by: Linu Cherian <lcherian@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/cgx.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/cgx.c b/drivers/soc/marvell/octeontx2/cgx.c
index 54b5e1622b72..0f3093dd66f8 100644
--- a/drivers/soc/marvell/octeontx2/cgx.c
+++ b/drivers/soc/marvell/octeontx2/cgx.c
@@ -581,7 +581,6 @@ static int cgx_fwi_link_change(struct cgx *cgx, int lmac_id, bool enable)
 
 	return cgx_fwi_cmd_generic(&req, &rsp, cgx, lmac_id);
 }
-EXPORT_SYMBOL(cgx_fwi_link_change);
 
 static inline int cgx_fwi_read_version(struct cgx_ver_s *ver, struct cgx *cgx)
 {
@@ -604,6 +603,9 @@ static int cgx_lmac_verify_fwi_version(struct cgx *cgx)
 	struct device *dev = &cgx->pdev->dev;
 	int err;
 
+	if (!cgx->lmac_count)
+		return 0;
+
 	err = cgx_fwi_read_version(&ver, cgx);
 	dev_dbg(dev, "Firmware command interface version = %d.%d\n",
 		ver.major_ver, ver.minor_ver);
-- 
2.17.1

